webapp/src/components/system_console/embedding_search_config.tsx

```diff
@@ -0,0 +1,623 @@
+// See LICENSE.txt for license information.
+
+import React, {useState, useEffect, useCallback} from 'react';
+import {useIntl, FormattedMessage} from 'react-intl';
+import styled from 'styled-components';
+
+import {Pill} from '../pill';
+import {PrimaryButton, SecondaryButton} from '../assets/buttons';
+import ConfirmationDialog from '../confirmation_dialog';
+import {doReindexPosts, getReindexStatus, cancelReindex} from '../../client';
+
+import {useIsBasicsLicensed} from '@/license';
+
+import {ItemList, SelectionItem, SelectionItemOption, TextItem, HelpText, ItemLabel} from './item';
+import Panel from './panel';
+import EnterpriseChip from './enterprise_chip';
+
+interface UpstreamConfig {
+    type: string;
+    parameters: Record<string, unknown>;
+}
+
+interface ChunkingOptions {
+    chunkSize: number;
+    chunkOverlap: number;
+    minChunkSize: number;
+    chunkingStrategy: string;
+}
+
+export interface EmbeddingSearchConfig {
+    type: string;
+    vectorStore: UpstreamConfig;
+    embeddingProvider: UpstreamConfig;
+    parameters: Record<string, unknown>;
+    dimensions: number;
+    chunkingOptions?: ChunkingOptions;
+}
+
+interface Props {
+    value: EmbeddingSearchConfig;
+    onChange: (config: EmbeddingSearchConfig) => void;
+}
+
+const Horizontal = styled.div`
+    display: flex;
+    flex-direction: row;
+    align-items: center;
+    gap: 8px;
+`;
+
+const ButtonContainer = styled.div`
+    margin-top: 24px;
+    padding-top: 24px;
+    border-top: 1px solid rgba(var(--center-channel-color-rgb), 0.08);
+    grid-column: 1 / -1;
+`;
+
+const ActionContainer = styled.div`
+    display: grid;
+    grid-template-columns: minmax(auto, 275px) 1fr;
+    grid-column-gap: 16px;
+`;
+
+const SuccessHelpText = styled(HelpText)`
+    margin-top: 8px;
+    color: var(--online-indicator);
+`;
+
+const ErrorHelpText = styled(HelpText)`
+    margin-top: 8px;
+    color: var(--error-text);
+`;
+
+const ProgressContainer = styled.div`
+    margin-top: 8px;
+    width: 100%;
+    background-color: rgba(var(--center-channel-color-rgb), 0.08);
+    border-radius: 4px;
+    height: 8px;
+    overflow: hidden;
+`;
+
+const ProgressBar = styled.div<{progress: number}>`
+    height: 100%;
+    width: ${(props) => props.progress}%;
+    background-color: var(--button-bg);
+    transition: width 0.3s ease-in-out;
+`;
+
+const ProgressText = styled(HelpText)`
+    margin-top: 8px;
+    margin-bottom: 12px;
+    font-size: 12px;
+`;
+
+const ButtonGroup = styled.div`
+    display: flex;
+    gap: 8px;
+`;
+
+// Match the server's JobStatus struct field names
+interface JobStatusType {
+    status: string; // 'running' | 'completed' | 'failed' | 'canceled' | 'no_job'
+    error?: string;
+    started_at: string; // ISO string from server's time.Time
+    completed_at?: string;
+    processed_rows: number;
+    total_rows: number;
+}
+
+const EmbeddingSearchPanel = ({value, onChange}: Props) => {
```

Comment:
nit: This component looks like a good candidate to be splitter in smaller components.